qrenderdoc/Windows/TextureViewer.cpp

```diff
@@ -3944,40 +3953,58 @@ void TextureViewer::on_cancelTextureListFilter_clicked()
   ui->textureListFilter->setCurrentText(QString());
 }
 
-void TextureViewer::on_textureListFilter_editTextChanged(const QString &text)
+void TextureViewer::on_colSelect_clicked()
 {
-  TextureListItemModel *model = (TextureListItemModel *)ui->textureList->model();
-
-  if(model == NULL)
-    return;
+  QStringList headers;
+  for(int i = 0; i < TextureListFilter::Column_Count; ++i)
+  {
+    headers.push_back(
+        ui->textureList->model()->headerData(i, Qt::Horizontal, Qt::DisplayRole).toString());
+  }
 
-  model->reset(TextureListItemModel::String, text);
+  UpdateVisibleColumns(tr("Select Texture List Columns"), TextureListFilter::Column_Count,
+                       ui->textureList->header(), headers);
 }
 
-void TextureViewer::on_textureListFilter_currentIndexChanged(int index)
+void TextureViewer::on_textureListFilter_editTextChanged(const QString &text)
 {
-  refreshTextureList();
+  refreshTextureList(FilterType::String, text);
 }
 
-void TextureViewer::refreshTextureList()
+void TextureViewer::on_textureListFilter_currentIndexChanged(int index)
 {
-  TextureListItemModel *model = (TextureListItemModel *)ui->textureList->model();
-
-  if(model == NULL)
-    return;
-
   if(ui->textureListFilter->currentIndex() == 1)
-    model->reset(TextureListItemModel::Textures, QString());
+    refreshTextureList(FilterType::Textures, QString());
   else if(ui->textureListFilter->currentIndex() == 2)
-    model->reset(TextureListItemModel::RenderTargets, QString());
+    refreshTextureList(FilterType::RenderTargets, QString());
   else
-    model->reset(TextureListItemModel::String, ui->textureListFilter->currentText());
+    refreshTextureList(FilterType::String, ui->textureListFilter->currentText());
```

Comment:
I'm not sure I understand.  I will change FilterType to be an `enum class` but it's just defined in TextureViewer.h.  Which names are you worried about collisions with?

Reply:
Yeh `FilterType` is local to TextureViewer but if it's a plain enum then `String` becomes valid in any class member function. That means it could override a `String` token at global scope - it doesn't leak out but it risks colliding with anything global.

Without it being an `enum class` then the `FilterType::` is optional at best (I actually thought it was illegal but this seems to compile), so generally for this kind of thing I prefer enum class in new code unless casting to/from ints is going to be a real chore.

Reply:
Got it, thanks!